### PR TITLE
[#44] Measure performance improvement of memoized severity.

### PR DESCRIPTION
**Pull request by 365andreas**
description
-----------

Measure the improvement of performance using the hashmap created to query the severity of each scribe combining the global minimum severity, scribe's minimum severity and trace's minimum severity 

checklist
---------

- [x] compiles (`cabal new-clean; cabal new-build`)
- [x] tests run successfully (`cabal new-test`)
- [x] documentation added and created (`cd docs; nix-shell --run make`)
- [x] link to an issue
- [x] link to an epic
- [x] add estimate points
- [x] add milestone (the same as the linked issue)

### CODE REVIEW

**Comment by CodiePP** [comment]
can't get it up to speed! 
```
iohk-monitoring
  Testing Trace
    stress testing: Memoized severity filter: FAIL (0.05s)
      test/Cardano/BM/Test/Trace.lhs:276:
      Memoized severity filter consumed more time than the original [29719,17588]
```

File: src/Cardano/BM/Trace.lhs [hunk 1]
```diff
@@ -233,14 +229,15 @@ traceConditionally
...
-    minsev <-liftIO $ getSeverity (configuration ctx) (minSeverity ctx) (loggerName ctx)
-#else
-    globminsev <- liftIO $ Config.minSeverity (configuration ctx)
-    globnamesev <- liftIO $ Config.inspectSeverity (configuration ctx) (loggerName ctx)
-    let minsev =
-            max (minSeverity ctx) $ max globminsev $ fromMaybe Debug globnamesev
-#endif
+    let conf = configuration ctx
+    memoizeSeverity <- liftIO $ getMemoizeSeverity conf
+    minsev <- liftIO $ if memoizeSeverity
```
Comment (CodiePP) [inline src/Cardano/BM/Trace.lhs]: this changes behaviour! `getSeverity` will test all paths up to the root, while the other direct functions will only evalute for the given logger name.

**Comment by 365andreas** [comment]
@CodiePP Should we close-abort this?

**Comment by CodiePP** [comment]
yes let's close this one. the memoization is being reverted by PR #135

**Pull request closed without merging**